### ISSUE TITLE
Question of loss functions used for fine-tuning the MLLM.

### ISSUE THREAD

**Issue by PureBinaryHeart** [niladridutt/monetGPT#2]
Hi there! Thanks for sharing this amazing work on MonetGPT. I've been going through the paper and trying to understand the implementation details.

I noticed that the paper doesn't explicitly mention the loss functions used for fine-tuning the MLLM. I'm curious about the specifics of how the loss is calculated during this process.

Could you please clarify what specific loss functions are used for each of the puzzles (A, B, and C)? Are they standard loss functions (e.g.,  MSE) ?

**Comment by niladridutt** [comment]
Hi @PureBinaryHeart 

Thanks for your interest! We use cross entropy loss in a supervised fine tuning setup. 
When we say the puzzles act like proxy loss functions for the image operations we mean the MLLM doesn't need to learn the image operation and back propagate through it which is hard for some image filters which may not be differentiable. Instead by learning to solve the puzzles, the MLLM learns the image operations as it needs to properly justify the reasoning. Hope this clarifies your doubt.

**Comment by tejank10** [comment]
Hi @niladridutt, great work! Continuing this thread for completeness, I am unsure of what is meant by regression in the context of this statement: "Additionally, the MLLM uses chain-of-thought (CoT) reasoning to regress from highlevel reasoning to precise parameter values, referencing a predefined legend that maps adjustments to numerical ranges" (sentence in the first para on page 6). Could you please clarify on what is meant by regression in this context? Thanks!
